internal/venice-common/src/main/java/com/linkedin/venice/pubsub/adapter/kafka/consumer/ApacheKafkaConsumerConfig.java

```diff
@@ -53,6 +79,126 @@ public ApacheKafkaConsumerConfig(VeniceProperties veniceProperties, String consu
     consumerProperties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class);
   }
 
+  public static Properties preparePubSubSSLPropertiesForServer(VeniceProperties veniceProperties) {
```

Comment:
Integration tests are failing due to this change so assume this is still under WIP? 

```
    java.lang.NullPointerException
	at com.linkedin.davinci.kafka.consumer.AggKafkaConsumerService.getCommonKafkaConsumerPropertiesForLocalIngestion(AggKafkaConsumerService.java:125)
```

Reply:
Regarding "point in the execution flow where configs are extracted",  the output properties of `PubSubXXAdapterConfig` will carry all the properties from server or controller instead of creating new empty properties, the method you point out is to add/change some fields. So I still think it serves the purpose. 

Of course, this is a big change, we need to be very careful about this. I discussed with @huangminchn, and we believe some kafka related configs might be shared with our internal logic, it may not be good to be put in the module of `venice.putsub.adapter.kafka`.  

To unblock our internal test, I think we can focus on this simple change first: https://github.com/linkedin/venice/pull/553. And then revisit this one. What do you think?

Reply:
>  the output properties of PubSubXXAdapterConfig will carry all the properties from server or controller instead of creating new empty properties,

No it won't. Take a look at `ApacheKafkaProducerConfig` and see how it works